src/backend/distributed/utils/multi_partitioning_utils.c

```diff
@@ -996,11 +996,20 @@ IsParentTable(Oid relationId)
 	systable_endscan(scan);
 	table_close(pgInherits, AccessShareLock);
 
-	if (tableInherited && PartitionedTable(relationId))
+	Relation relation = try_relation_open(relationId, AccessShareLock);
+	if (relation == NULL)
+	{
+		ereport(ERROR, (errmsg("could not create distributed table: "
```

Comment:
this function is not specific to create_distributed_table, maybe just "relation with OID %d does not exist"